tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1191,6 +1214,36 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
       return true;
     }
 
+    if (gemm->custom_call_target() == kCublasLtMatmulF8CallTarget &&
+        bias->shape().element_type() == F32 && convert != nullptr) {
+      HloInstruction *bias_f16_maybe = convert->mutable_operand(0);
+      auto compatible_bias_type = [](const PrimitiveType btype,
+                                     const PrimitiveType dtype) {
+        if (btype == BF16) {
+          return dtype == F8E4M3FN || dtype == F8E5M2 || dtype == F32 ||
+                 dtype == BF16;
+        } else if (btype == F16) {
+          return dtype == F16 || dtype == F8E4M3FN || dtype == F8E5M2;
+        }
+        return false;
+      };
+
+      // cuBLAS LT does not support FP32 biases on matmuls with FP8 inputs,
+      // even if the matmul output is FP32. We do not unconditionally convert
+      // the bias to a supported precision (F16 or BF16) because this lowers
+      // precision. Instead, we only fuse the bias if the bias itself is a
+      // convert from F16 or BF16, fusing the input of the convert instruction
+      // to the matmul.
+      if (compatible_bias_type(bias_f16_maybe->shape().element_type(),
+                               gemm->shape().element_type())) {
+        bias = bias_f16_maybe;
+      } else {
+        VLOG(1) << "Epilogue fusion of FP32 vector bias into FP8 GEMM is "
+                   "currently not supported. See the cublasLT support matrix.";
+        return true;
```

Comment:
Shouldn't this return false?

Reply:
Here returning true means: don't continue to fuse as matrix bias. That's a difference b/w non-fp8 matmul. Because in FP8, matrix bias is not implemented in `FuseMatrixBiasAdd` but in `CreateF8CustomCall` which precedes `HandelAdd`.

Reply:
But https://github.com/tensorflow/tensorflow/pull/60319 changes it so `HandleAdd` handles a matrix bias. And the return value of this function is assigned to to a local variable called `was_fused`. Even without https://github.com/tensorflow/tensorflow/pull/60319, it wont' cause any harm to return false here (and on line 1207), and makes it more clear.